presto-iceberg/src/test/java/com/facebook/presto/iceberg/BenchmarkHadoopCatalogOptimization.java

```diff
@@ -0,0 +1,115 @@
+
+import java.util.OptionalInt;
+
+import static com.facebook.presto.iceberg.CatalogType.HADOOP;
+import static com.facebook.presto.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.apache.iceberg.FileFormat.PARQUET;
+import static org.openjdk.jmh.annotations.Mode.AverageTime;
+import static org.openjdk.jmh.annotations.Scope.Thread;
+
+@State(Thread)
+@OutputTimeUnit(MILLISECONDS)
+@BenchmarkMode(AverageTime)
+@Fork(2)
+@Warmup(iterations = 10)
+@Measurement(iterations = 10)
+public class BenchmarkHadoopCatalogOptimization
+{
+    @State(Thread)
+    public static class Context
+    {
+        private DistributedQueryRunner hadoopQueryRunner;
+
+        public final DistributedQueryRunner getHadoopQueryRunner()
+        {
+            return hadoopQueryRunner;
+        }
+
+        @Setup
+        public void setUp()
+                throws Exception
+        {
+            hadoopQueryRunner = createIcebergQueryRunner(
+                    ImmutableMap.of(),
+                    ImmutableMap.of("iceberg.catalog.type", HADOOP.name()),
+                    PARQUET,
+                    false,
+                    true,
+                    OptionalInt.empty());
+
+            hadoopQueryRunner.execute("CREATE SCHEMA iceberg.test_hadoop");
+            hadoopQueryRunner.execute("CREATE TABLE iceberg.test_hadoop.region AS SELECT * from tpch.tiny.region");
+            hadoopQueryRunner.execute("CREATE TABLE iceberg.test_hadoop.orders AS SELECT * from tpch.tiny.orders");
+            hadoopQueryRunner.execute("CREATE TABLE iceberg.test_hadoop.lineitem AS SELECT * from tpch.tiny.lineitem");
+        }
+
+        @TearDown
+        public void tearDown()
+        {
+            hadoopQueryRunner.execute("DROP TABLE IF EXISTS iceberg.test_hadoop.region");
+            hadoopQueryRunner.execute("DROP TABLE IF EXISTS iceberg.test_hadoop.orders");
+            hadoopQueryRunner.execute("DROP TABLE IF EXISTS iceberg.test_hadoop.lineitem");
+            hadoopQueryRunner.execute("DROP SCHEMA IF EXISTS iceberg.test_hadoop");
+            hadoopQueryRunner.close();
+            hadoopQueryRunner = null;
+        }
+    }
+
+    @Benchmark
+    public MaterializedResult benchmarkIcebergTableSelect(Context context)
+    {
+        return context.getHadoopQueryRunner()
+                .execute("SELECT * FROM iceberg.test_hadoop.region");
+    }
+
+    @Benchmark
+    public MaterializedResult benchmarkIcebergTableTpchQ4(Context context)
+    {
+        return context.getHadoopQueryRunner()
+                .execute("SELECT o.orderpriority, count(*) AS order_count FROM iceberg.test_hadoop.orders o WHERE o.orderdate >= DATE '1993-07-01' AND o.orderdate < DATE '1993-07-01' + INTERVAL '3' MONTH AND EXISTS ( SELECT * FROM iceberg.test_hadoop.lineitem l WHERE l.orderkey = o.orderkey AND l.commitdate < l.receiptdate) GROUP BY o.orderpriority ORDER BY o.orderpriority");
```

Comment:
Is it possible to format the lines using SqlFormatter e.g. [Instant SQL Formatter & Beautifier Online](https://www.dpriver.com/pp/sqlformat.htm) ?